### PR TITLE
for-pullreqwest

### PR DESCRIPTION
**Pull request by israeljacob**

### CODE REVIEW

**Review by talbarfm — CHANGES_REQUESTED** [review]
.

File: communicating_vessels.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+def interleave(*organs):
+    # Create an empty list to store the interleaved elements
```
Comment (talbarfm) [inline communicating_vessels.py]: I would use documentation instead, comments make it look messy
https://swimm.io/blog/code-comments-vs-documentation/

File: communicating_vessels.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    my_list = []
+    # Find the length of the longest argument
+    max_organ = 0
+    for organ in organs:
+        if len(organ) > max_organ:
+            max_organ = len(organ)
```
Comment (talbarfm) [inline communicating_vessels.py]: In python you can use the max function

File: communicating_vessels.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    # Loop through each index from 0 to max_organ
+    for i in range(0, max_organ):
+        # Loop through each argument and append the i-th element to my_list
+        for organ in organs:
```
Comment (talbarfm) [inline communicating_vessels.py]: Rename 'i'

File: cup_of_join.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+def join(*lists, sep="-"):
```
Comment (talbarfm) [inline cup_of_join.py:1]: I would use documantation instead, comments make it look messy
https://swimm.io/blog/code-comments-vs-documentation/

There's no need for the blank lines

File: cup_of_join.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    for list1 in lists:
+        # Loop through each item in the current list
+        for organ in list1:
+            # If the item is another list, recursively call the join function
+            # with the nested list and separator as arguments and append the result to new_list
+            if type(organ) == list:
+                new_list.append(join(organ, sep))
+            # If the item is not a list, directly append it to new_list
+            else:
+                new_list.append(organ)
```
Comment (talbarfm) [inline cup_of_join.py]: 1. Python is snake case (list1)-
https://www.google.com/amp/s/www.theserverside.com/answer/Camel-case-vs-snake-case-Whats-the-difference%3famp=1

2. Replace for loops -  
https://medium.com/python-pandemonium/never-write-for-loops-again-91a5a4c84baf
Replace for loops with more advanced python features

File: cup_of_join.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        # Append the separator to new_list after all the items in the current list have been appended
+        new_list.append(sep)
+
+    # Remove the last separator appended to new_list and return new_list
+    new_list.pop()
+    return new_list
+
+
+# Call the join function with a single list containing the integer 1 as its only item and print the result
+print(join([1]))
```
Comment (talbarfm) [inline cup_of_join.py]: Use _name==main_ - 
https://docs.python.org/3/library/__main__.html
https://www.geeksforgeeks.org/python-main-function/amp/

File: piece_of_cake.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
+def get_recipe_price(prices, optionals=[], **products):
```
Comment (talbarfm) [inline piece_of_cake.py:1]: I would use documantation instead, comments make it look messy
https://swimm.io/blog/code-comments-vs-documentation/

File: thats_the_way.py [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import os
+
+def find_deep_files(path):
+    deep_files = []
+    for file in os.listdir(path):
+        if file.startswith("deep"):
+            deep_files.append(file)
+    return deep_files
```
Comment (talbarfm) [inline thats_the_way.py]: You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.

File: thats_the_way.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+path = "C:\\Users\\yisra\\Downloads\\Notebooks-master\\week05\\images"
+
+if os.path.exists(path):
+    deep_files = find_deep_files(path)
+    print(deep_files)
+else:
+    print(f"The path {path} does not exist.")
```
Comment (talbarfm) [inline thats_the_way.py]: Separate code to small functions- 
https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be

File: whisperer.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
+def messages(path):
```
Comment (talbarfm) [inline whisperer.py:1]: I would use documentation instead, comments make it look messy
https://swimm.io/blog/code-comments-vs-documentation/

File: whisperer.py [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    # Open the file located at the given path in binary mode
+    with open(path, 'rb') as f:
```
Comment (talbarfm) [inline whisperer.py]: rename 'f'

File: whisperer.py [hunk 12]
```diff
@@ -0,0 +1,31 @@
...
+                    # Reset the message string to an empty string
+                    message = ""
+        # Close the file
+        f.close()
+
+# Prompt the user to enter a path for a file
+path = input("Enter the path of the picture\n")
+# Call the messages function with the provided path and assign the resulting generator object to the ret variable
+ret = messages(path)
+# Iterate over each message in the ret generator object and print it to the console
```
Comment (talbarfm) [inline whisperer.py]: Use _name==main_ - 
https://docs.python.org/3/library/__main__.html
https://www.geeksforgeeks.org/python-main-function/amp/

**Review by talbarfm — CHANGES_REQUESTED** [review]
.

File: communicating_vessels.py (hunk 1 above)
Comment (israeljacob) [reply communicating_vessels.py]: I fixed it

File: communicating_vessels.py (hunk 2 above)
Comment (israeljacob) [reply communicating_vessels.py]: I fixed it

File: communicating_vessels.py (hunk 3 above)
Comment (israeljacob) [reply communicating_vessels.py]: I fixed it

File: cup_of_join.py (hunk 4 above)
Comment (israeljacob) [reply cup_of_join.py:1]: I fixed it

File: piece_of_cake.py (hunk 7 above)
Comment (israeljacob) [reply piece_of_cake.py:1]: I fixed it

File: whisperer.py (hunk 10 above)
Comment (israeljacob) [reply whisperer.py:1]: I fixed it

File: cup_of_join.py (hunk 5 above)
Comment (israeljacob) [reply cup_of_join.py]: I fixed it

File: cup_of_join.py (hunk 6 above)
Comment (israeljacob) [reply cup_of_join.py]: I fixed it

File: thats_the_way.py (hunk 8 above)
Comment (israeljacob) [reply thats_the_way.py]: I fixed it

File: thats_the_way.py (hunk 9 above)
Comment (israeljacob) [reply thats_the_way.py]: I fixed it

File: whisperer.py (hunk 11 above)
Comment (israeljacob) [reply whisperer.py]: I fixed it

File: whisperer.py (hunk 12 above)
Comment (israeljacob) [reply whisperer.py]: I fixed it

**Pull request merged**